map.js

```diff
@@ -14,90 +19,149 @@ let ctx;
 let zoomLevel = 1;
 let minZoom = 0.2;
 let maxZoom = 3.5;
-let mapX = 0;
-let mapY = 0;
+
+/**
+ * map coordinate offset from screen coords
+ */
+let mapOffset = new Point(0,0);
+let iconH = 1;
 let currentOverlay = "Locations"; // Locations, NirnRoute, Exploration.
-let hoverOverlayButton = 0;
 let hoverLocation = "";
+
+//image objects
+let topbar;
+let overlay;
+
+/**
+ * Last position of the mouse. used for rendering mouseover stuff.
+ */
+var lastMouseLoc = new Point(0,0);
 let mousedown = false;
 
 let img_Map;
 let icons = {};
 
-//TODO: These all need to be reworked to just use the JsonData trees
-let locArr = [];
-let nirnArr = [];
-let discoveredArr = []; //I know you probably hate me for adding ANOTHER array to fix later.
+/**
+ * An object that will be displayed on the map canvas.
+ */
+function MapObject(){
+    this.minX = 0;
+    this.maxX = 0;
+    this.minY = 0;
+    this.maxY = 0;
+}
+
+MapObject.prototype.contains = function(point){
+    return (this.minX < point.x && point.x < this.maxX && this.minY < point.y && point.y < this.maxY);
+}
+MapObject.prototype.width = function(){
+    return this.maxX - this.minX;
+}
+MapObject.prototype.height = function(){
+    return this.maxY - this.minY;
+}
 
 async function initMap(){
     //load map cord data
-    
-    //TODO: remove locArr and nirnArr and just use the jsondata trees
-    loadJsonData().then(()=>{
-        runOnTree(jsondata.nirnroot, x=>{if(x.cell == "Outdoors")nirnArr.push(x)});
-        runOnTree(jsondata.location, x=>locArr.push(x));
-    });
-    
-       
-    //do we still need to do this if the map is on its own page?
-    //TODO: create window here 
-    //TODO: do hide n seek stuff
 
     viewport = document.getElementById("wrapper_Map");
-    canvas = document.getElementById("canvas_Map");
+
+    canvas = document.createElement("CANVAS");
+    canvas.id = "canvas_Map";
+    canvas.width = 3544;
+    canvas.height = 2895;
+    viewport.appendChild(canvas);
     ctx = canvas.getContext("2d");
-    
     await initImgs();
+    initTopbar();
+    initOverlay();
     initListeners();
 
     //center map on imp city
-    mapX = 1700;
-    mapY = 885;
+    mapOffset = new Point(1700,885);
+
+    drawFrame();
+    console.log("map init'd");
+}
 
-    drawMap();
+function drawFrame(){
+    drawBaseMap();
+    drawMapOverlay();
+    //TODO: don't have topbar overlay map. or move topbar or something aaa idk
```

Comment:
We could make the map topbar just be a separate html element right above it, and link the buttons from there to map.js.